File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        if(!list1 && !list2) return nullptr;
        ListNode* curr = new ListNode();
        ListNode* res = curr;

        while(list1 && list2) {
            if(list1->val < list2->val) {
                res->val = list1->val;
                list1 = list1->next;
            } else {
                res->val = list2->val;
                list2 = list2->next;
            }
            if(list1 || list2){
                res->next = new ListNode();
                res = res->next;
            }
        }

        while(list1) {
            res->val = list1->val;
            list1 = list1->next;
            if(list1) {
                res->next = new ListNode();
                res = res->next;
            }
        }

        while(list2) {
            res->val = list2->val;
            list2 = list2->next;
            if(list2) {
                res->next = new ListNode();
                res = res->next;
            }
        }

        return curr;
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        res = ListNode()
        copy = res

        while l1 and l2:
            if l1.val < l2.val:
                res.next = ListNode(l1.val)
                res = res.next

                l1 = l1.next
            else:
                res.next = ListNode(l2.val)
                res = res.next

                l2 = l2.next
        
        while l1:
            res.next = ListNode(l1.val)
            res = res.next

            l1 = l1.next

        while l2:
            res.next = ListNode(l2.val)
            res = res.next

            l2 = l2.next
        return copy.next